pep-0604.rst

```diff
@@ -74,61 +76,72 @@ Simplified Syntax
     assert str | int == Union[str,int]
     assert str | int | float == Union[str, int, float]
 
-    # Instead of
-    # typing.List[typing.Union[str, int]]
+    # Instead of typing.List[typing.Union[str, int]]
     typing.List[str | int]
 
-    # list[typing.Union[str, int]] (Currently TypeError: 'type' object has no attribute '__getitem__')
-    list[int | str]
-
     # Instead of typing.Dict[str, typing.Union[int, float]]
-    typing.Dict[str, int | float]
     dict[str, int | float]
+    list[str | int]
 
-The new syntax should be accepted for calls to `isinstance` as long as the Union items are valid arguments to `isinstance` themselves.
-
-::isinstance
-::
-    # valid
-    isinstance("", int | str)
-
+The existing ``typing.Union`` and ``|`` syntax should be equivalent.
 
-    # invalid
-    isinstance(2, list[int]) # TypeError: isinstance() argument 2 cannot be a parameterized generic
-    isinstance(1, int | list[int])
-
-::subclass
 ::
-    # valid
-    assert issubclass(bool, int | float)
 
-    # invalid
-
-The existing `typing.Union` and `|` syntax should be equal.
-::
   int | str == typing.Union[int, str]
 
   typing.Union[int, int] == int
   int | int == int
 
 The order of the items in the Union should not matter for equality.
+
 ::
+
   (int | str) == (str | int)
   (int | str | float) == typing.Union[str, float, int]
 
 
-Optional values should be equal to the new union syntax, for example:
+Optional values should be equivalentl to the new union syntax
+
 ::
+
   None | t == typing.Optional[t]
 
-A new Union.__str__() method should be implemented.
+A new Union.__repr__() method should be implemented.
+
 ::
-  (int | list[str]).__str__()
+
+  str(int | list[str])
   # int | list[str]
 
-  (int | int).__str__()
+  str(int | int)
   # int
 
+isinstance
+----------
+
+The new syntax should be accepted for calls to ``isinstance`` as long as the Union items are valid arguments to ``isinstance`` themselves.
+
+::
+
+    # valid
+    isinstance("", int | str)
+
+    # invalid
+    isinstance(2, list[int]) # TypeError: isinstance() argument 2 cannot be a parameterized generic
+    isinstance(1, int | list[int])
+
+issubclass
+----------
+The new syntax should be accepted for calls to ``issubclass`` as long as the Union items are valid arguments to ``issubclass`` themselves.
+
+::
+
+  # valid
+  assert issubclass(bool, int | float)
+
+  # invalid
+  assert issubclass(bool, bool | list[int])
+
```

Comment:
There's a lot of duplication for isinstance and subclass -- can you combine the two sections?